src/hotspot/os/linux/osContainer_linux.cpp

```diff
@@ -137,9 +137,9 @@ jlong OSContainer::pids_current() {
   return cgroup_subsystem->pids_current();
 }
 
-void OSContainer::print_container_helper(outputStream* st, jlong j, const char* metrics) {
+void OSContainer::print_container_helper(outputStream* st, jlong j, const char* metrics, boolean limit) {
   st->print("%s: ", metrics);
-  if (j > 0) {
+  if ((j > 0) || ((j == 0) && (limit == false))) {
```

Comment:
Clearer would be a reversal like this:
```
if (j == 0 && limit) {
  "unlimited"
} else {
  print
}
```

Reply:
I think I decided to have it this way to draw the attention to the fact that the value `0` for a `non-limit` kind should be treated as numerical value and not a special case.

This provides, in my opinion, a self-documenting code without the need for an explicit comment. That was my rationale at least, but if that doesn't come across, then we can simplify and add an actual comment?

Reply:
Up to you. Both are okay, this was just a small nit :)